### PR TITLE
Tracing support with custom metrics collection

### PR DESCRIPTION
**Pull request by bonclay7**
### What does this PR do?

**1/ Export traces to AWS X-Ray for the EKS infrastructure monitoring (experimental).**

This deploys the operator in a daemon set mode (a collector per instance). and exposes local HTTP and gRPC ports for applications to send traces to X-Ray. Feature is set to experimental for evaluation

**2/ (bonus) Allows custom metrics collection within infra module**

Allow collecting metrics for any given ports, dashboards are not provided.

```hcl
  # additional custom metrics
  enable_custom_metrics = true
  custom_metrics_config = {
    ports                   = [8000, 8080]
    dropped_series_prefixes = ["go_gcc"]
  }
```

image

--
Docs pages will come in a separate PR

Issue #27

### CODE REVIEW

File: modules/workloads/infra/otel-config/values.yaml [hunk 1]
```diff
@@ -4,40 +4,11 @@ ekscluster: ${eks_cluster}
 
 globalScrapeTimeout: ${global_scrape_timeout}
 globalScrapeSampleLimit: ${global_scrape_sample_limit}
-# TODO: enable after terraform 1.3 as defaults will be optional
 
-#nodeExporterScrapeInterval: ${node_exporter_scrape_interval}
-#nodeExporterScrapeTimeout: ${node_exporter_scrape_timeout}
+enableTracing: ${enable_tracing}
+xRayGrpcEndpoint: ${xray_grpc_endpoint}
```
Comment (awsimaya) [inline modules/workloads/infra/otel-config/values.yaml:9]: Can you change the variable name to otlpGrpcEndpoint? Same applies to the HttpEndpoint too.

**Comment by awsimaya** [comment]
LGTM, except the one comment I left for the variable name.

**Review by lewinkedrs — APPROVED** [review]
LGTM!

**Comment by bonclay7** [comment]
Thanks for your review. I have a few test items I need to check before merging

File: modules/workloads/infra/otel-config/templates/opentelemetrycollector.yaml [hunk 2]
```diff
@@ -3,19 +3,50 @@ kind: OpenTelemetryCollector
...
+      {{ end }}
       prometheus:
         config:
           global:
             scrape_interval: {{ .Values.globalScrapeInterval }}
             scrape_timeout: {{ .Values.globalScrapeTimeout }}
             external_labels:
               cluster: {{ .Values.ekscluster }}
-              account_id: {{ .Values.accountId }}
+              # account_id: {{ .Values.accountId }}
```
[the remark this answers is not in the document]
Comment (bonclay7) [reply modules/workloads/infra/otel-config/templates/opentelemetrycollector.yaml]: Good catch!

File: modules/workloads/infra/otel-config/values.yaml (hunk 1 above)
Comment (bonclay7) [reply modules/workloads/infra/otel-config/values.yaml]: Good catch, TY!

**Pull request merged**